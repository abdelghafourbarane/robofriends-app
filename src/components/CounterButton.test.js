import { shallow } from "enzyme";
import React from "react";

import CounterButton from "./CounterButton";

describe("tests Counter Button component", () => {
  it("expect to render CounterButton component", () => {
    expect(shallow(<CounterButton />)).toMatchSnapshot();
  });

  it("expect to increment the counter", () => {
    const colorMock = "red";
    const wrapper = shallow(<CounterButton color="red" />);
    wrapper.find('[id="button"]').simulate("click");
    expect(wrapper.state()).toEqual({ count: 2 });
    expect(wrapper.props().color).toEqual(colorMock);
  });
});
