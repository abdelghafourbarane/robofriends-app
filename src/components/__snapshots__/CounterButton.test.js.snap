// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Counter Button component expect to render CounterButton component 1`] = `ShallowWrapper {}`;
